src/base/static/components/place-detail/index.js

```diff
@@ -210,6 +215,90 @@ class PlaceDetail extends Component {
     });
   }
 
+  onAdditionalData(action, payload) {
+    switch (action) {
+      case constants.ON_ADD_ATTACHMENT_ACTION:
+        this.attachments.push(payload);
+        break;
+      default:
+        console.error(
+          "Error: Unable to handle form field callback action:",
+          action
+        );
+        break;
+    }
+  }
+
+  onGeometryStyleChange(style) {
+    this.geometryStyle = style;
+  }
+
+  onEditorUpdate(fieldState) {
```

Comment:
Would we still want to keep the Backbone-specific stuff in `PlaceDetail`? If so I don't think there's a lot we can move down to the `PlaceDetailEditor` component. The alternative would be, as you say, to pass the model itself down, but I've become fond of keeping the model logic in one place as high as possible in the hierarchy.

It makes for a large `PlaceDetail` component, but perhaps that's just a symptom of the Backbone-React friction, which will eventually be factored out?

Reply:
Instead of passing the Backbone model down to `PlaceDetailEditor`, can we instead pas a prop like `onModelSave` that calls `this.props.model.save`? 

That should give `PlaceDetailEditor` the ability to trigger a save on the model instance, which we can stub out for a future refactor of the Backbone models.

ie:

```
<PlaceDetailEdtior
    onModelSave: {this.props.model.save}
    ...
/>
```

Reply:
Oh right-- I guess this fits in with the pattern of only passing down the parts of an object's API that we actually need in child components.